test/e2e/route_service_test.go

```diff
@@ -0,0 +1,81 @@
+// +build e2e
+
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"testing"
+
+	"knative.dev/serving/pkg/apis/serving/v1alpha1"
+	"knative.dev/serving/test"
+	v1a1test "knative.dev/serving/test/v1alpha1"
+	"knative.dev/pkg/test/logstream"
+	"knative.dev/pkg/apis"
+)
+
+// type containerOption func(*corev1.Container)
+
+var serviceCondSet = apis.NewLivingConditionSet(
+	v1alpha1.ServiceConditionConfigurationsReady,
+	v1alpha1.ServiceConditionRoutesReady,
+)
+
+// TestRouteNotReady tests the scenario that when route's status is
+// Ready == false, the service's RouteReady value should change from
+// Unknown to False
+func TestRouteNotReady(t *testing.T) {
+	t.Parallel()
+	cancel := logstream.Start(t)
+	defer cancel()
+
+	clients := Setup(t)
+
+	names := test.ResourceNames{
+		Service: test.ObjectNameForTest(t),
+		Image:   "foobar",  // This is an invalid image, which will cause the creaetd route to be not ready
+	}
+
+	test.CleanupOnInterrupt(func() { test.TearDown(clients, names) })
+	defer test.TearDown(clients, names)
+
+	t.Log("Creating a new Service with RouteReady == Unknown and route Ready == False")
+	err := v1a1test.CreateRunLatestServiceNotReady(t, clients, &names, &v1a1test.Options{
```

Comment:
You should be able to just use `CreateLatestService` here. You wait on line 67 for RouteReady to become false there doesn't seem like a need to wait until Service ready becomes False first.

Reply:
Yeah we could use `CreateLatestService`, but will have to wait 10 minutes for it to timeout waiting for the service to become ready https://github.com/knative/serving/blob/master/test/v1alpha1/service.go#L128
https://github.com/knative/serving/blob/master/test/v1alpha1/configuration.go#L46